### PR TITLE
Legacy importer for  groups + positions

### PR DESCRIPTION
**Pull request by koivyr**
### Summary
- Legacy importer for groups, houses, and positions
- also changed the group name and position name column sizes. 

### Test Plan
-

### CODE REVIEW

**Comment by koivyr** [comment]
I'm not sure why make test is hanging...???

File: donut/modules/groups/utils/house_importer.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import csv
+
+
+def import_file(env, filename):
+    # Create reference to DB
+    db = make_db(env)
+    try:
+        db.begin()
+        with open(filename, "r") as f:
+            reader = csv.DictReader(f, delimiter=",")
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: I think `delimeter=','` is the default

File: donut/modules/groups/utils/house_importer.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            for i, line in enumerate(reader):
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: Don't need `i`

File: donut/modules/groups/utils/house_importer.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import argparse
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: `directory_search/utils/import_legacy.py` already handles house membership imports, do we need this?

File: donut/modules/groups/utils/house_importer.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    return None if res is None else res['pos_id']
+
+
+if __name__ == "__main__":
+    # Parse input arguments
+    parser = argparse.ArgumentParser(
+        description='Imports a list of past house members from the old db')
+    parser.add_argument(
+        "-e", "--env", default="dev", help="Database to update")
+    parser.add_argument("file", help="Path to old media wiki xml")
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: Update this description; which file is it?

File: donut/modules/groups/utils/house_importer.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    with db.cursor() as cursor:
+        cursor.execute(query, [uid])
+        res = cursor.fetchone()
+    return None if res is None else res['user_id']
+
+
+def get_position_id(member_type, house, db):
+    query = "SELECT pos_id FROM positions NATURAL JOIN groups WHERE group_name = %s and pos_name = %s"
+    with db.cursor() as cursor:
+        cursor.execute(query, [house + " House", member_type])
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: Group name should just be the house name, sorry. I fixed this in #135.

File: donut/modules/groups/utils/house_importer.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+from donut.pymysql_connection import make_db
+import csv
+
+
+def import_file(env, filename):
+    # Create reference to DB
+    db = make_db(env)
+    try:
+        db.begin()
+        with open(filename, "r") as f:
```
Comment (calebsander) [inline donut/modules/groups/utils/house_importer.py]: `'r'` is the default

File: donut/modules/groups/utils/position_importer.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                """
+                with db.cursor() as cursor:
+                    cursor.execute(query,
+                                   [pos_id, user_id, start_date, end_date])
+        db.commit()
+    finally:
+        db.close()
+
+
+def get_create_group_id(group_name, db):
```
Comment (calebsander) [inline donut/modules/groups/utils/position_importer.py:36]: Let's not reimport the houses; they will already have been imported by the `directory_search` legacy importer

File: donut/modules/groups/utils/position_importer.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+def get_group_id(group_name, db):
+    group_id_query = "SELECT group_id FROM groups WHERE group_name=%s"
+    with db.cursor() as cursor:
+        cursor.execute(group_id_query, [group_name])
+        res = cursor.fetchone()
+    return res
+
+
+def clean_names(group_name):
+    return group_name.replace("Hovse", "House")
```
Comment (calebsander) [inline donut/modules/groups/utils/position_importer.py]: lol

File: donut/modules/groups/utils/position_importer.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    return None if res is None else res['user_id']
+
+
+def get_create_position_id(pos_name, group_id, control, db):
+    insertion_query = "INSERT INTO positions(group_id, pos_name, control) VALUES(%s, %s, %s)"
+    res = get_position_id(pos_name, group_id, db)
+    if res is None:
+        with db.cursor() as cursor:
+            cursor.execute(insertion_query, [
+                group_id, pos_name, True if control == "t" else False
```
Comment (calebsander) [inline donut/modules/groups/utils/position_importer.py]: `True if control == "t" else False` is just `control == 't'`

File: donut/modules/groups/utils/position_importer.py (hunk 8 above)
Comment (calebsander) [reply donut/modules/groups/utils/position_importer.py]: House group names should just be the house name (without `House` or `Hovse`), sorry. I fixed this in #135.

File: donut/modules/groups/utils/position_importer.py [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    # Parse input arguments
+    parser = argparse.ArgumentParser(
+        description='Imports a list of past house members from the old db.')
+    # SELECT start_date, end_date,  pos_name , org_name, uid, control
+    # FROM position_holders NATURAL JOIN position_titles NATURAL JOIN
+    # position_organizations NATURAL JOIN inums NATURAL JOIN undergrads;
+
+    parser.add_argument(
+        "-e", "--env", default="dev", help="Database to update")
+    parser.add_argument("file", help="Path to old media wiki xml")
```
Comment (calebsander) [inline donut/modules/groups/utils/position_importer.py]: Which file does this need to be? How are you getting the file? From the `directory_search` legacy export?

**Review by calebsander — CHANGES_REQUESTED** [review]

**Comment by calebsander** [comment]
RE the tests not being collected in time: add `donut/modules/groups/utils` to the ignored directories in `pytest.ini`. (The symlink creates a cycle in the directory graph, so pytest sees an infinite set of files to explore.)

File: donut/modules/groups/utils/house_importer.py (hunk 3 above)
Comment (koivyr) [reply donut/modules/groups/utils/house_importer.py]: Woops didn’t realize >.<

File: donut/modules/groups/utils/position_importer.py (hunk 9 above)
Comment (koivyr) [reply donut/modules/groups/utils/position_importer.py]: ._. don't code in high stress and minimum sleep

File: donut/modules/groups/utils/position_importer.py (hunk 10 above)
Comment (koivyr) [reply donut/modules/groups/utils/position_importer.py]: it's the query that i had in the comments

File: donut/modules/groups/utils/position_importer.py [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        db.begin()
+        with open(filename) as f:
+            reader = csv.DictReader(f)
+            for line in reader:
+                uid = line['uid']
+                group_name = clean_names(line['org_name'])
+                pos_name = line['pos_name']
+                control = line['control']
+                start_date = line['start_date']
+                end_date = line['end_date']
```
Comment (calebsander) [inline donut/modules/groups/utils/position_importer.py:19]: I don't think most of our code currently checks whether the position is currently active. I'll work on actually using this information.

**Review by calebsander — APPROVED** [review]

**Pull request merged**